components/square/sources/new-order-created/new-order-created.mjs

```diff
@@ -0,0 +1,67 @@
+import base from "../common/base.mjs";
+import constants from "../common/constants.mjs";
+
+export default {
+  ...base,
+  key: "square-new-order-created",
+  name: "New Order Created",
+  description: "Emit new event for every new order created",
+  type: "source",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    ...base.props,
+    location: {
+      propDefinition: [
+        base.props.square,
+        "location",
+      ],
+    },
+  },
+  hooks: {
+    ...base.hooks,
+    async deploy() {
+      console.log(`Retrieving at most last ${constants.MAX_HISTORICAL_EVENTS} objects...`);
+      const response = await this.square.listOrders({
+        params: {
+          limit: constants.MAX_HISTORICAL_EVENTS,
+        },
+        data: {
+          query: {
+            sort: {
+              sort_field: "CREATED_AT",
+              sort_order: "DESC",
+            },
+          },
+          location_ids: [
+            this.location,
+          ],
+        },
+      });
+      response?.objects?.slice(-constants.MAX_HISTORICAL_EVENTS)
```

Comment:
this instruction is repeated over the sources and could be reused by a inherited method